include/string.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright 2020 Merzlov Nikolay merzlovnik@mail.ru
+#include "string.hpp"
+
+String::~String() {
+    delete[] Data;
+}
+
+String::String() : size(0) {
+    Data = new char[size];
+}
+
+String::String(const String &rhs) : size(rhs.size) {
+    Data = new char[size];
+    for (unsigned int i = 0; i < size; ++i) {
+        Data[i] = rhs[i];
+    }
+}
+
+String::String(const char *data) : size(0) {
+    const char *copy_str = data;
+    for (unsigned int i = 0; copy_str[i] != '\000'; ++copy_str)
+        ++size;
+    Data = new char[size];
+    for (unsigned int i = 0; i < size; ++i)
+        Data[i] = data[i];
+}
+
+void String::Resize(size_t new_size) {
+    String copy_str(*this);
+    delete[]Data;
+    Data = new char[new_size];
+    for (unsigned int i = 0; i < copy_str.size; ++i)
+        Data[i] = copy_str.Data[i];
+}
+
+String &String::operator=(const String &rhs) {
+    if (&rhs != this) {
+        Data = new char[rhs.size];
+        size = rhs.size;
+        for (unsigned int i = 0; i < size; ++i)
+            Data[i] = rhs[i];
+    }
+    return *this;
+}
+
+class String &String::operator+=(const class String &rhs) {
+    size = size + rhs.size;
+    Resize(size);
+    for (unsigned int i = size - rhs.size, j = 0; i < size + 1; ++i, ++j)
+        Data[i] = rhs.Data[j];
+    return *this;
+}
+
+class String &String::operator+=(const char *rhs) {
+    String str(rhs);
+    size = size + str.size;
+    Resize(size);
+    for (unsigned int i = size - str.size, j = 0; i < size; ++i, ++j)
+        Data[i] = str.Data[j];
+    return *this;
+}
+
+class String &String::operator*=(unsigned int m) {
+    String copy_str(*this);
+    for (unsigned int i = 1; i < m; ++i)
+        *this += copy_str;
+    return *this;
+}
+
+bool String::operator==(const class String &rhs) const {
+    if (size != rhs.size)
+        return false;
+    if (size == rhs.size) {
+        for (unsigned int i = 0; i < size; ++i) {
+            if (Data[i] != rhs.Data[i])
+                return false;
+        }
+    }
+    return true;
+}
+
+bool operator==(const char *ch, const String &str) {
+    String rhs(ch);
+    if (str.Size() != rhs.Size())
+        return false;
+    if (str.Size() == rhs.Size()) {
+        for (unsigned int i = 0; i < str.Size(); ++i) {
+            if (str[i] != rhs[i])
+                return false;
+        }
+    }
+    return true;
+}
+
+bool String::operator<(const class String &rhs) const {
+    for (unsigned int i = 0; i < rhs.size && i < size; ++i) {
+        if (Data[i] > rhs.Data[i])
+            return false;
+        if (Data[i] < rhs.Data[i])
+            return true;
+    }
+    return size < rhs.size;
+}
+
+size_t String::Find(const class String &substr) const {
+    unsigned int j = 0;
+    size_t sub_size = 0, sub_pos = 0, last_pos = -1;
+    for (unsigned int i = 0; i < size; ++i) {
```

Comment:
There is no need to iterate up to the last symbol, think which index is the last possible.